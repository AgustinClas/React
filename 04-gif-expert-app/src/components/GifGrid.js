import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    const {loading, data:images} = useFetchGifs(category);

    return (
        <div className = "GifGrid">
            <h3>{category}</h3>
            {loading && <p>Cargando</p>} 
            <div className='card-grid'>
                {
                    images.map((img) =>(
                        <GifGridItem 
                        {...img}
                        key = {img.id}
                        />
                        ))
                }
            </div> 
        </div>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}