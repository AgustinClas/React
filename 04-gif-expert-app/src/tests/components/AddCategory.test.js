import React, { useState } from "react";
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import { AddCategory } from "../../components/AddCategory";

describe('Pruebas en AddCategory', () => {

    const setCategories = jest.fn();

    let wrapper;

    beforeEach(() => {

        jest.clearAllMocks();

        wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    
    });
    
    test('Debe mostrar los elementos de la manera correspondiente', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    /* test('Debe de cambiar la caja de texto', () => {
      
        const input = wrapper.find('input');
      
        input.simulate('change', {target: {value : "Hola"}} );
        
    }); */
    
    
    test('No debe implementar setCategories', () => {
        
        wrapper.find('form').simulate( 'submit', { preventDefault(){} } );
        
        expect( setCategories ).not.toHaveBeenCalled
    });
    
    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        
        const input = wrapper.find('input');
      
        input.simulate('change', {target: {value : "Hola"}} );

        wrapper.find('form').simulate( 'submit', { preventDefault(){} } );

        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledWith(expect.any(Function));
        expect( input.prop('value').trim()).toBe('');


    });
    
    
  
});
