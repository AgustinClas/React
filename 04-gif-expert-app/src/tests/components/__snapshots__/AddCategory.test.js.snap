// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory Debe mostrar los elementos de la manera correspondiente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    id="1"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
